src/code42cli/cmds/cases.py

```diff
@@ -0,0 +1,250 @@
+@sdk_options()
+def update(state, case_number, name, subject, assignee, description, notes, status):
+    """Update case details for the given case."""
+    state.sdk.cases.update(
+        case_number,
+        name=name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+        status=status,
+    )
+
+
+@cases.command("list")
+@click.option(
+    "--name", help="Filter by name of a case, supports partial name matches.",
+)
+@click.option("--subject", help="Filter by user UID of the subject of a case.")
+@click.option("--assignee", help="Filter by user UID of assignee.")
+@click.option(
+    "--begin-create-time",
+    help="Fetch cases created after given date time.",
+    type=click.DateTime(formats=[DATE_FORMAT]),
+)
+@click.option(
+    "--end-create-time",
+    help="Fetch cases created before given date time.",
+    type=click.DateTime(formats=[DATE_FORMAT]),
+)
+@click.option(
+    "--begin-update-time",
+    help="Fetch cases last updated after given date time.",
+    type=click.DateTime(formats=[DATE_FORMAT]),
+)
+@click.option(
+    "--end-update-time",
+    help="Fetch cases last updated before given date time.",
+    type=click.DateTime(formats=[DATE_FORMAT]),
+)
+@click.option("--status", help="Filter cases by case status.")
+@format_option
+@sdk_options()
+def _list(
+    state,
+    name,
+    assignee,
+    subject,
+    begin_create_time,
+    end_create_time,
+    begin_update_time,
+    end_update_time,
+    status,
+    format,
+):
+    """List all the cases."""
+    pages = state.sdk.cases.get_all(
+        name=name,
+        assignee=assignee,
+        subject=subject,
+        min_create_time=begin_create_time,
+        max_create_time=end_create_time,
+        min_update_time=begin_update_time,
+        max_update_time=end_update_time,
+        status=status,
+    )
+    formatter = OutputFormatter(format, _get_cases_header())
+    cases = [case for page in pages for case in page["cases"]]
+    if cases:
+        formatter.echo_formatted_list(cases)
+    else:
+        click.echo("No cases found.")
+
+
+@cases.command()
+@case_number_arg
+@sdk_options()
+@format_option
+def show(state, case_number, format):
+    """Show case details."""
+    formatter = OutputFormatter(format, _get_cases_header())
+    try:
+        response = state.sdk.cases.get(case_number)
+        formatter.echo_formatted_list([response.data])
+    except Py42NotFoundError:
+        click.echo("Invalid case-number {}.".format(case_number))
+
+
+@cases.command()
+@case_number_arg
+@click.option(
+    "--path", type=str, help="File path. Defaults to current directory.", default="."
+)
+@sdk_options()
+def export(state, case_number, path):
+    """Download case detail summary in a pdf file at the given path with name <case_number>_case_summary.pdf."""
+    response = state.sdk.cases.export_summary(case_number)
+    with open("{}/{}_case_summary.pdf".format(path, case_number), "wb") as f:
```

Comment:
On a `Windows` machine, using a path with the other slashes `\`, will this work?

Reply:
also should we check whether the last character is a path separator or not, a user may or may not pass it!

Reply:
I think if you do `os.path.join()`, it will automatically take care of that.